src/app/course/courses-list/courses-list.component.html

```diff
@@ -0,0 +1,4 @@
+<ng-container>
+    <app-course-item *ngFor="let course of courses" [course]="course" (actionPerformed)="actionPerformedOnItem($event)"></app-course-item>
```

Comment:
also, you can try track by for ngFor.
Is usefull enough for working with large number of items.
You can take a look at
https://medium.com/better-programming/improving-angular-ngfor-performance-through-trackby-ae4cf943b878